src/browser/browser.js

```diff
@@ -50,8 +51,14 @@ define((require, exports, module) => {
 
   const modifier = OS.platform() == 'linux' ? 'alt' : 'accel';
   const KeyDown = KeyBindings({
-    'accel l': _ => Input.Action({action: Focusable.Focus()}),
-    'accel t': _ => WebView.Action({action: WebView.Open()}),
+    'accel l': _ => Input.Action({
+      action: Focusable.Focus(),
+      source: 'keyboard'
```

Comment:
Not entirely sure where the `source` field was introduced but I'd recommend against it as with it we end up with multi-level polymorphism. What I mean by that is we use types to annotate a raw data, if we introduce `source` field it kind of serves same purpose but in a different style which means we'll have two types of annotations to check to decide how to act. What I propose to de instead is define a different type for that for example `Input.KeyboardAction({action})` that way you'll be able to distinguish between actions triggered by keyboard or differently by checking `instanceof`.

I don't know enough context here to say for sure but in general I'd recommend to avoid distinction by a triggering source and instead use of natural terms for naming actions. For example we could have `EditLocation` action & `TransitionToLocationEdit` action that would transition state to the same mode but with a different animation.

I think that is more inline with a desired effect because `source: 'keyboard'` does not necessary have to translate to quick / slow animation. Hypothetically we make have another keyboard shortcut say `accept shift t` that  behaves identically to clicking **+** button in which case encoding source of an action won't be able to handle it.